### PR TITLE
docs: document late binding tstops

### PR DESCRIPTION
**Pull request by AayushSabharwal**
## Checklist

- [ ] Appropriate tests were added
- [ ] Any code changes were done in a way that does not break public API
- [ ] All documentation related to code changes were updated
- [ ] The new code follows the
  [contributor guidelines](https://github.com/SciML/.github/blob/master/CONTRIBUTING.md), in particular the [SciML Style Guide](https://github.com/SciML/SciMLStyle) and
  [COLPRAC](https://github.com/SciML/COLPRAC).
- [ ] Any new documentation only uses public API

### CODE REVIEW

File: src/solve.jl [hunk 1]
```diff
@@ -800,7 +800,8 @@ section at the end of this page for some example usage.
   multistep methods), then `tstops` will use an interpolation,
   matching the behavior of `saveat`. If a method cannot change timesteps and
   also cannot interpolate, then `tstops` must be a multiple of `dt` or else an
-  error will be thrown. Default is `[]`.
+  error will be thrown. `tstops` may also be a function accepting the parameter
```
Comment (ChrisRackauckas) [inline src/solve.jl:803]:
```suggestion
  error will be thrown. `tstops` may also be a function `tstops(p, tspan)`, accepting the parameter
```

**Comment by ChrisRackauckas** [comment]
If this is not implemented in DiffEqBase, then we need to make a trait for it, and trait check its usage in DiffEqBase so it's only given to appropriate solvers.

**Comment by AayushSabharwal** [comment]
I guess it doesn't matter, since we can't implement the late binding tstops here if we can't guarantee/check if `add_tstop!` is valid.

**Comment by AayushSabharwal** [comment]
Trait defined in https://github.com/SciML/SciMLBase.jl/pull/858

**Pull request merged**